### ISSUE TITLE
index.android.bundle is readable and understandable, But Facebook apk does not contain any  index.android.bundle

### ISSUE THREAD

**Issue by Louies89** [facebook/react-native#22934]
**index.android.bundle** is just a  uglified and minified js file. and it can be beautified and code is understandable. As facebook claims that facebook app is made by react native, then why there is no index.android.bundle in facebook apk?

If there, facebook javascript code are nicely taken care, so that third party can not access it.

If facebook is using some kind of tool, then why its not available for react-native users ? Without some kind of security, our app and business logic becomes vulnerable.

**Comment by juangl** [comment]
There's no way to prevent your app from being reverse-engineered even if it's binary. See for instance: https://ibotpeaches.github.io/Apktool/.

**Comment by evertonbuild** [comment]
@Louies89 

I noticed that the facebook application has bundle files in .hbc format. I opened the file, but I saw that it's some kind of binary.
This same type of file is also found inside the Instagram apk.

So I went to check the apk of other companies that also use react-native as technology to create their applications, but within are normally index.android.bundle found.

I would like to know which encryption tool Facebook and Instagram are using and if their documentation for the developer community is available.
